### PR TITLE
Feat/form validation

### PR DESCRIPTION
**Pull request by thomas-mauran**
## What does this PR do ?

Creates form validation for frontend register and login forms 

### How should this be manually tested?

- Step 1 : clone and run
- Step 2 : try the forms

### Other changes
- Adding Yup and Formik lib

### Boyscout

### CODE REVIEW

File: src/pages/RegisterPage.tsx [hunk 1]
```diff
@@ -1,6 +1,24 @@
+import { RegisterFormType } from '@/typings/form.type';
 import { Box, Button, Checkbox, Flex, FormControl, FormLabel, Heading, Input, Stack } from '@chakra-ui/react';
+import { Formik, Form, Field, ErrorMessage } from 'formik';
+import * as Yup from 'yup';
 
 export const RegisterPage = () => {
+  const validationSchema = Yup.object({
+    username: Yup.string().required('Username is required'),
+    email: Yup.string().email('Invalid email address').required('Email is required'),
```
Comment (mfernd) [inline src/pages/RegisterPage.tsx]: add `.max(100)`

File: src/pages/RegisterPage.tsx [hunk 2]
```diff
@@ -1,6 +1,24 @@
...
+    password: Yup.string().required('Password is required').min(6, 'Password must be at least 6 characters'),
```
Comment (mfernd) [inline src/pages/RegisterPage.tsx]: add `max(255)`

File: src/typings/form.type.ts
Comment (mfernd) [inline src/typings/form.type.ts:1]: so interface or type?

Comment (thomas-mauran) [reply src/typings/form.type.ts:1]: image
Interface today

File: src/pages/RegisterPage.tsx (hunk 2 above)
Comment (thomas-mauran) [reply src/pages/RegisterPage.tsx]: done :+1:

File: src/pages/RegisterPage.tsx (hunk 1 above)
Comment (thomas-mauran) [reply src/pages/RegisterPage.tsx]: done :+1:

**Review by noetarbouriech — APPROVED** [review]

**Pull request merged**